### PR TITLE
chore: allows Quay to run for account recoveries (PROJQUAY-970)

### PR DESCRIPTION
**Pull request by kleesc**
Adds ACCOUNT_RECOVERY_MODE to allow Quay to run with some core
features disabled. When this is set, the instance should only be used
in order by existing users who hasn't linked their account to an
external login service, after database authentication has been
disabled.

### CODE REVIEW

**Comment by syed** [comment]
Is it possible to add a warning on the UI that says something along the lies of this is a recovery-only endpoint. Push/Pull and other APIs will not work

**Review by syed — DISMISSED** [review]

**Comment by syed** [comment]
I did a simple test locally pushing and pulling images with recovery enabled/disabled

 ### `ACCOUNT_RECOVERY_MODE: true`

pushes fail as expected

```
[parallels@localhost local-dev]$ podman push --tls-verify=false localhost:8080/admin/traefik
Getting image source signatures
Error: Error copying image to the remote destination: Error trying to reuse blob sha256:6169bf830ae69f192382cfe9a6fe96e7f28a21c0c5e5045b85871b510de58435 at destination: failed to read from destination repository admin/traefik: 405 (Method Not Allowed)
```

pulls fail as expected

```
[parallels@localhost local-dev]$ podman pull --tls-verify=false localhost:8080/admin/traefik
Trying to pull localhost:8080/admin/traefik...
  error parsing HTTP 405 response body: no error details found in HTTP response body: "{\"error\": \"Quay running for account recoveries only.\"}"
Error: unable to pull localhost:8080/admin/traefik: Error initializing source docker://localhost:8080/admin/traefik:latest: Error reading manifest latest in localhost:8080/admin/traefik: error parsing HTTP 405 response body: no error details found in HTTP response body: "{\"error\": \"Quay running for account recoveries only.\"}"
[parallels@localhost local-dev]$ 
```

 ### `ACCOUNT_RECOVERY_MODE: false`


```
[parallels@localhost local-dev]$ podman push --tls-verify=false localhost:8080/admin/traefik
Getting image source signatures
Copying blob 7e3a7faa369b skipped: already exists  
Copying blob a8ffa77de707 skipped: already exists  
Copying blob 6169bf830ae6 skipped: already exists  
Copying blob b40cf96be96e [--------------------------------------] 0.0b / 0.0b
Copying config 746e8d5c61 [======================================] 4.2KiB / 4.2KiB
Writing manifest to image destination
Storing signatures
```

```
[parallels@localhost local-dev]$ podman pull --tls-verify=false localhost:8080/admin/traefik
Trying to pull localhost:8080/admin/traefik...
Getting image source signatures
Copying blob 72825cc0e363 skipped: already exists  
Copying blob bbe4dc4ed4e1 skipped: already exists  
Copying blob d85aa6467d90 skipped: already exists  
Copying blob 1c22f933917a [--------------------------------------] 0.0b / 0.0b
Copying config 746e8d5c61 done  
Writing manifest to image destination
Storing signatures
746e8d5c613a5c8b9102e79f6c859e1e0876f50e263a62c45cae30ecace0a14c
```

both pulls and pushes work

**Review by syed — DISMISSED** [review]

File: static/directives/quay-message-bar.html [hunk 1]
```diff
@@ -1,10 +1,17 @@
-<div class="announcement inline quay-message-bar-element" ng-show="messages.length || inReadOnlyMode">
+<div class="announcement inline quay-message-bar-element" ng-show="messages.length || inReadOnlyMode || inAccountRecoveryMode">
   <div class="quay-service-status-description info" ng-if="inReadOnlyMode">
     <div style="display: inline-block">
       <span class="registry-name"></span>&nbsp;is currently in read-only mode. Pulls and other read-only operations
       will succeed but all other operations are currently suspended.
     </div>
   </div>
+  <div class="quay-service-status-description info" ng-if="inAccountRecoveryMode">
```
Comment (syed) [inline static/directives/quay-message-bar.html:8]: Can we increase the severity from `info` to `warning`

**Review by syed — CHANGES_REQUESTED** [review]

File: static/directives/quay-message-bar.html (hunk 1 above)
Comment (kleesc) [reply static/directives/quay-message-bar.html]: Done

**Review by syed — APPROVED** [review]

**Pull request merged**